docs/adr/0005-support-multi-environments-in-repos.md

```diff
@@ -0,0 +1,91 @@
+# Repositories supporting multiple environment deployments
+
+## Context and Problem Statement
+
+With the introduction of the [app of apps](0004-argocd-apps-of-apps-structure.md) ADR we now have a way to separate application
+management with infrastructure management. The [argocd-apps](https://github.com/operate-first/argocd-apps) repo now serves
+as a central point of managing ArgoCD applications (and in turn our manifests in general) for the operate-first initiative.
+
+As per the [App of Apps ADR doc](0004-argocd-apps-of-apps-structure.md) we would like both the `argocd-apps` repo and 
+the repos that each application in the `argocd-apps` target to be able to be deployed in multiple clusters. 
+
+For example, I should be able to create a single `ArgoCD` app, that points to the `argocd-apps` repo and have it create
+all the ArgoCD applications that live in this repo, regardless of cluster/environment. This is fine if your manifest
+repo only has one set of manifest deployment that can be deployed in any environment without needing additional tuning.
+This is rarely the case however, and often times you'll have a `common` set of manifests, and then your target environments
+will add additional configurations on top of this. This results in multiple subdirectories with different `paths` for 
+each target cluster / environment. This essentially means we can only use `argocd-apps` repo for a specific cluster only.
+
+
+## Considered Options
+1) Keep argocd-apps repo cluster specific, and use multiple repos as "app-of-apps" repos for each cluster/environment
+2) Use Kustomize overlays/bases for argocd-apps repo and have different overlays that patch the `path` for each cluster/environment
+
+## Decision Outcome
+Choose 2) as it accomplishes much of 1) without needing yet more repos and adding bloat to the organization. 
+
+The `argocd-apps` repo would follow the following structure: 
+
+The example makes these assumptions: 
+- There are 2 target clusters: `cluster_1` and `cluster_2` 
```

Comment:
I'm a little confused by "cluster", isnt it simply an "environment", that could be two environments running on the same OpenShift cluster?!

Reply:
Yeah it can, I'll change cluster to environments.

Reply:
updated to be more general: changed cluster -> environments